uaclient/files/notices.py

```diff
@@ -0,0 +1,211 @@
+import os
+from collections import namedtuple
+from enum import Enum
+from typing import List, Tuple
+
+from uaclient import defaults, event_logger, messages, system
+
+event = event_logger.get_event_logger()
+NoticeFileDetails = namedtuple(
+    "NoticeFileDetails", ["order_id", "label", "is_permanent", "message"]
+)
+
+
+class Notice(NoticeFileDetails, Enum):
+    REBOOT_REQUIRED = NoticeFileDetails(
+        label="reboot_required",
+        order_id="10",
+        is_permanent=False,
+        message="System reboot required",
+    )
+    ENABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="enable_reboot_required",
+        order_id="11",
+        is_permanent=False,
+        message=messages.ENABLE_REBOOT_REQUIRED_TMPL,
+    )
+    REBOOT_SCRIPT_FAILED = NoticeFileDetails(
+        label="reboot_script_failed",
+        order_id="12",
+        is_permanent=True,
+        message=messages.REBOOT_SCRIPT_FAILED,
+    )
+    FIPS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_reboot_required",
+        order_id="20",
+        is_permanent=False,
+        message=messages.FIPS_REBOOT_REQUIRED_MSG,
+    )
+    FIPS_SYSTEM_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_system_reboot_required",
+        order_id="21",
+        is_permanent=False,
+        message=messages.FIPS_SYSTEM_REBOOT_REQUIRED.msg,
+    )
+    FIPS_INSTALL_OUT_OF_DATE = NoticeFileDetails(
+        label="fips_install_out_of_date",
+        order_id="22",
+        is_permanent=True,
+        message=messages.FIPS_INSTALL_OUT_OF_DATE,
+    )
+    FIPS_DISABLE_REBOOT_REQUIRED = NoticeFileDetails(
+        label="fips_disable_reboot_required",
+        order_id="23",
+        is_permanent=False,
+        message=messages.FIPS_DISABLE_REBOOT_REQUIRED,
+    )
+    FIPS_PROC_FILE_ERROR = NoticeFileDetails(
+        label="fips_proc_file_error",
+        order_id="24",
+        is_permanent=True,
+        message=messages.FIPS_PROC_FILE_ERROR,
+    )
+    FIPS_MANUAL_DISABLE_URL = NoticeFileDetails(
+        label="fips_manual_disable_url",
+        order_id="25",
+        is_permanent=True,
+        message=messages.NOTICE_FIPS_MANUAL_DISABLE_URL,
+    )
+    WRONG_FIPS_METAPACKAGE_ON_CLOUD = NoticeFileDetails(
+        label="wrong_fips_metapackage_on_cloud",
+        order_id="25",
+        is_permanent=True,
+        message=messages.NOTICE_WRONG_FIPS_METAPACKAGE_ON_CLOUD,
+    )
+    LIVEPATCH_LTS_REBOOT_REQUIRED = NoticeFileDetails(
+        label="lp_lts_reboot_required",
+        order_id="30",
+        is_permanent=False,
+        message=messages.LIVEPATCH_LTS_REBOOT_REQUIRED,
+    )
+    CONTRACT_REFRESH_WARNING = NoticeFileDetails(
+        label="contract_refresh_warning",
+        order_id="40",
+        is_permanent=True,
+        message=messages.NOTICE_REFRESH_CONTRACT_WARNING,
+    )
+    OPERATION_IN_PROGRESS = NoticeFileDetails(
+        label="operation_in_progress",
+        order_id="60",
+        is_permanent=True,
```

Comment:
this can be false (just to be defensive in case of errors - there is no case where this file would be valid across a reboot)